Data/AuthenticationClient/Tests/AuthentincationDataManagerTests.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  AuthentincationDataManagerTests.swift
+//  QuranEngine
+//
+//  Created by Mohannad Hassan on 26/12/2024.
+//
+
+import AppAuth
+import Combine
+import Foundation
+import XCTest
+@testable import AuthenticationClient
+
+final class AuthentincationDataManagerTests: XCTestCase {
+    // MARK: Internal
+
+    let configuration = OAuthAppConfiguration(
+        clientID: "client-id",
+        redirectURL: URL(string: "callback://")!,
+        scopes: [],
+        authorizationIssuerURL: URL(string: "https://example.com")!
+    )
+
+    override func setUp() {
+        caller = OAuthCallerMock()
+        persistance = PersistanceMock()
+        sut = AuthentincationDataManagerImpl(caller: caller, persistance: persistance)
+    }
+
+    func testNoConfigurations() async throws {
+        XCTAssertEqual(sut.authenticationState, .notAvailable, "Expected to signal a not-configured state")
+        do {
+            try await sut.login(on: UIViewController())
+            XCTFail("Expected to throw an error if prompted to login without app configuration being set.")
+        } catch {
+            // Success
+        }
+    }
+
+    func testLoginSuccessful() async throws {
+        sut.set(appConfiguration: configuration)
+
+        persistance.currentState = AuthenticationData()
+
+        let state = AutehenticationDataMock()
+        state.accessToken = "abcd"
+        caller.loginResult = .success(state)
+
+        do {
+            try await sut.login(on: UIViewController())
+
+            XCTAssertTrue(persistance.clearCalled, "Expected to clear the persistance first")
+            XCTAssertEqual((persistance.currentState as? AutehenticationDataMock), state, "Expected to update the new state")
+            XCTAssertEqual(sut.authenticationState, .authenticated, "Expected the auth manager to be in authenticated state")
+        } catch {
+            XCTFail("Expected to login successfully -- \(error)")
+        }
```

Comment:
The do-catch is not needed. XCTest will fail the test if an except is thrown. Same in other tests below